broadinstitute/pearl
# AR-111 adding oh basic survey

This adds the basic info survey for OurHealth.  Along the way, it adds two important pieces of infrastructure:
1. AnswerMappings, which allow mapping of survey questions to data model fields (currently only the Profile is supported), so that, e.g., a survey asking the participant's name can be populated back to their profile.  (see the entry on AnswerMappings at https://broadworkbench.atlassian.net/wiki/spaces/PEARL/pages/2693857309/Surveys)
2. DataChangeRecords, which can be use to track any time a piece of data about a participant is changed.  Note that we log it to a table since these records will be PHI/PII and so we do not want to send that info to regular logs.

Now, when a survey response is received, the SnapshotProcessingService is invoked to handle any updates to participant profile that need to be performed as part of the survey.  Note that the processing that can be done now is extremely basic -- there's no validation and only text->string transforms are supported.  The goal was to get just enough done to get the concept in, and then later tickets (like AR-147) will handle enhancements.

TO TEST:
1. restart AdminApiApp so that migrations are run
2. restart ParticipantApiApp
3. go to `http://sandbox.ourhealth.localhost:3001/hub`
4. sign in as "consented@test.com", using the development mode sign-in
5. confirm "The basics" appears in your survey list
![image](https://user-images.githubusercontent.com/2800795/221685366-9a145420-2593-4566-b1e0-1281ab31c98b.png)
6. click "continue surveys"
7. fill out the survey, making a note of what you fill in as first and last names (but otherwise you can go as fast as you can -- we'll polish the content in later tickets)
8. after you complete the survey, go to `http://localhost:3000/ourhealth/studies/ourheart/env/sandbox/participants`
9. confirm that the participant with shortcode `OHSENT` now has the updated name you filled in.
